packages/core/src/PartialExecutionUtils/__tests__/DirectedGraph.test.ts

```diff
@@ -89,6 +90,105 @@ describe('DirectedGraph', () => {
 		});
 	});
 
+	describe('getStronglyConnectedComponents', () => {
+		// ┌─────┐    ┌─────┐    ┌─────┐
+		// │node1├───►│node2├───►│node4│
+		// └─────┘    └──┬──┘    └─────┘
+		//    ▲          │
+		//    │          │
+		// ┌──┴──┐       │
+		// │node3│◄──────┘
+		// └─────┘
+		test('find strongly connected components', () => {
+			// ARRANGE
+			const node1 = createNodeData({ name: 'Node1' });
+			const node2 = createNodeData({ name: 'Node2' });
+			const node3 = createNodeData({ name: 'Node3' });
+			const node4 = createNodeData({ name: 'Node4' });
+			const graph = new DirectedGraph()
+				.addNodes(node1, node2, node3, node4)
+				.addConnections(
+					{ from: node1, to: node2 },
+					{ from: node2, to: node3 },
+					{ from: node3, to: node1 },
+					{ from: node2, to: node4 },
+				);
+
+			// ACT
+			const stronglyConnectedComponents = graph.getStronglyConnectedComponents();
+
+			// ASSERT
+			expect(stronglyConnectedComponents).toHaveLength(2);
+			expect(stronglyConnectedComponents).toContainEqual([node4]);
+			expect(stronglyConnectedComponents).toContainEqual([node3, node2, node1]);
```

Comment:
I don't think we care about the order. It should be stable but it does not convey any information as far as I know, and as far as the use case is concerned.
So it's ok to leave it like this so that we don't have to touch the tests if we change the algorithm.